subsys/bluetooth/host/l2cap.c

```diff
@@ -3035,6 +3036,105 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 	return 0;
 }
 
+#if defined(CONFIG_BT_L2CAP_RECONFIGURE_EXPLICIT)
+int bt_l2cap_ecred_chan_reconfigure_explicit(struct bt_l2cap_chan **chans, uint16_t mtu,
+					     uint16_t mps)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct bt_l2cap_le_chan *ch;
+	struct net_buf *buf;
+	bool multiple_chan;
+	uint8_t ident;
+	int i;
+
+	LOG_DBG("chans %p mtu 0x%04x", chans, mtu);
+
+	if (!chans) {
+		return -EINVAL;
+	}
+
+	if (chans[0] == NULL) {
+		return -EINVAL;
+	}
+
+	if (mps > BT_L2CAP_RX_MTU) {
+		return -EINVAL;
+	}
```

Comment:
MTU range check? 

Is there a minimum MPS value?

Reply:
Good point. I think it's 23, but it's worded weirdly:

> L2CAP implementations shall support a minimum MPS of 23 octets and may support an MPS up to 65533 octets.

[Core 3.A.4.22 v6.0](https://www.bluetooth.com/wp-content/uploads/Files/Specification/HTML/Core-60/out/en/host/logical-link-control-and-adaptation-protocol-specification.html#:~:text=L2CAP%20implementations%20shall%20support%20a%20minimum%20MPS%20of%2023%20octets%20and%20may%20support%20an%20MPS%20up%20to%2065533%20octets.)

What does "support" mean? Did they mean to say simply "Allowed range is 23 to 65533."? I don't understand why the spec writes make everything so difficult.

Reply:
It is worded weirdly, but I think it's OK to assume that any values below 23 should be rejected. 

Do we have any checks if we receive a request from a peer with MPS < 23?

Reply:
23 is for LE Credits Based connections,  for Enhanced Credit Based Connections min MPS is 64 

also, being super strict on spec,  reconfiguration is procedure defined for Enhanced Credit Based only... 🫠

Reply:
> 23 is for LE Credits Based connections, for Enhanced Credit Based Connections min MPS is 64
> 
> also, being super strict on spec, reconfiguration is procedure defined for Enhanced Credit Based only... 🫠

64 it is then. AFAIK, we don't have any way to tell whether a `bt_l2cap_chan` is in LE Credit Based Flow Control mode or Enhanced Credit Based Flow Control mode, so I won't be adding a check in this PR. But we should make a TODO.